### ISSUE TITLE
Saving empty embedded collection of Message model does not work

### ISSUE THREAD

**Issue by lawmicha** [aws-amplify/amplify-ios#595]
**Describe the bug**
Using Datastore, saving a user object with empty or nil `messages` should be successful. Saving with a single message works.

**To Reproduce**
schema used:
```
type User
@model
{
  id: ID!
  messages: [Message]
}

type Message
{
  id: ID!
  sender: MessageSender!
  description: String!
  createdAt: AWSDateTime!
}
enum MessageSender {
  guest
  user
}
```

1. Save User with `[]` or `nil` messages and see the failure in logs about API. Local store is updated, mutation is not synced successfully to the cloud.
2. Save User with `Message` in the messsages array works.

**Environment(please complete the following information):**
- Amplify CLI 4.22.0
- Amplify Library - 1.0.3

**Comment by lawmicha** [comment]
1. AppSync endpoint with API key, no conflict resolution enabled.

AppSync console works:
- 
![image](https://user-images.githubusercontent.com/1365977/86055126-32a2e200-ba10-11ea-9593-4478eb6b7476.png)
![image](https://user-images.githubusercontent.com/1365977/86055186-42bac180-ba10-11ea-829a-0d5bdd3b44f3.png)

**Comment by lawmicha** [comment]
2. enabled conflict resolution, deleted data from DynamoDB

AppSync console fails with `[]`:

<img src="https://user-images.githubusercontent.com/1365977/86055932-777b4880-ba11-11ea-833c-fc49ec13d7ff.png" width="1080">

successfully with `null`
<img src="https://user-images.githubusercontent.com/1365977/86056165-da6cdf80-ba11-11ea-83f1-5bac0cd884e8.png" width="1080">

**Comment by lawmicha** [comment]
opening issue over in Amplify-CLI https://github.com/aws-amplify/amplify-cli/issues/4708

**Comment by lawmicha** [comment]
inside a DataStore app, I was able to leverage that passing in `null` works in the AppSync console, so this is sucessfully saving to cloud and I was able to verify it in the DynamoDB table
```
        let user = User(messages: nil)
        Amplify.DataStore.save(user) { (result) in
            switch result {
            case .success(let savedModel):
                self.message = "Saved Successfully \(savedModel)"
                print(self.message)
            case .failure(let error):
                self.message = "Failed to save \(error)"
                print(self.message)
            }
        }
```

**Comment by lawmicha** [comment]
this is pending service team

**Comment by drochetti** [comment]
This is still a relevant issue. AppSync is working with Amplify CLI on a fix.

**Comment by lawmicha** [comment]
i was able to verify that this is fixed with amplify 4.29.6
